tests/unit/datarobot_drum/drum/root_predictors/test_drum_server_utils.py

```diff
@@ -141,3 +152,161 @@ def test_calls_thread_correctly(self, mock_get_command, runner):
+
+        # Call the function
+        _queue_output(mock_stdout, mock_stderr, test_queue)
+
+        # Verify the queue contains all the expected output lines
+        expected_lines = [
+            b"stdout line 1\n",
+            b"stdout line 2\n",
+            b"stderr line 1\n",
+            b"stderr line 2\n",
+        ]
+
+        # Get all items from the queue and compare with expected
+        actual_lines = []
+        while not test_queue.empty():
+            actual_lines.append(test_queue.get())
+
+        assert actual_lines == expected_lines
+
+        # Verify close was called on both streams
+        mock_stdout.close.assert_called_once()
+        mock_stderr.close.assert_called_once()
+
+    @patch("builtins.print")
+    def test_stream_p_open_handles_process_output(self, mock_print):
+        # Create a mock Popen object
+        mock_process = Mock(spec=subprocess.Popen)
+
+        # Create real BytesIO objects that will be read by the real _queue_output function
+        mock_process.stdout = io.BytesIO(b"stdout line 1\nstdout line 2\n")
+        mock_process.stderr = io.BytesIO(b"stderr line 1\nstderr line 2\n")
+
+        # Configure process to terminate after first poll
+        mock_process.poll.side_effect = [None, 0]
+
+        # Call the function with real stdout/stderr streams
+        stdout, stderr = _stream_p_open(mock_process)
+
+        # Verify the results
+        assert stdout == ""
+        assert stderr == ""
+
+        # Verify print was called with the expected content
+        expected_outputs = [
+            b"stdout line 1",
+            b"stdout line 2",
+            b"stderr line 1",
+            b"stderr line 2",
+        ]
+
+        # Get the actual arguments passed to print
+        actual_outputs = [call_args[0][0] for call_args in mock_print.call_args_list]
+
+        # Verify all expected outputs were printed
+        assert len(actual_outputs) == len(expected_outputs)
+        for expected in expected_outputs:
+            assert expected in actual_outputs
+
+        # Verify poll was called to check for process termination
+        assert mock_process.poll.call_count == 2
+
+    @patch("builtins.print")
+    def test_stream_p_open_handles_empty_streams(self, mock_print):
+        # Create a mock Popen object with empty streams
+        mock_process = Mock(spec=subprocess.Popen)
+        mock_process.stdout = io.BytesIO(b"")
+        mock_process.stderr = io.BytesIO(b"")
+
+        # Configure process to terminate immediately
+        mock_process.poll.return_value = 0
+
+        # Call the function
+        stdout, stderr = _stream_p_open(mock_process)
+
+        # Verify results
+        assert stdout == ""
+        assert stderr == ""
+        mock_print.assert_not_called()  # No output to print
+        mock_process.poll.assert_called_once()  # Process termination was checked
+
+    @patch("builtins.print")
+    def test_stream_p_open_handles_delayed_output(self, mock_print):
```

Comment:
I am _very_ suspicious of this test.  

1. it seems very very delicate
2. it relies on timing and sleeps
3. it takes at least a tenth of a second to run which is way way too long for a unit test (longer because you have a one-seocond wait for the process to close).

this test points to a larger testability issue here.